composer/packages/ast-model/src/default-nodes/resources/function.json

```diff
@@ -1 +1,115 @@
\ No newline at end of file
+{
+  "ws": [
+    {
+      "ws": "\n\n",
+      "i": 7,
+      "text": "function",
+      "static": true
+    },
+    {
+      "ws": " ",
+      "i": 9,
+      "text": "name",
+      "static": false
+    },
+    {
+      "ws": "",
+      "i": 10,
+      "text": "(",
+      "static": true
+    },
+    {
+      "ws": "",
+      "i": 11,
+      "text": ")",
+      "static": true
+    },
+    {
+      "ws": " ",
+      "i": 13,
+      "text": "{",
+      "static": true
+    },
+    {
+      "ws": "\n\n",
+      "i": 15,
+      "text": "}",
+      "static": true
+    }
+  ],
+  "position": {
+    "startColumn": 1,
+    "startLine": 3,
+    "endColumn": 2,
+    "endLine": 5
+  },
+  "id": "236927f9-0d86-4f9c-aa92-86b1ff58e6e4",
+  "VisibleEndpoints": [],
+  "workers": [],
+  "returnTypeAnnotationAttachments": [],
+  "name": {
+    "position": {
+      "startColumn": 10,
+      "startLine": 3,
+      "endColumn": 14,
+      "endLine": 3
+    },
+    "id": "1176c111-9fe3-4b07-a62c-14c741ae93d1",
+    "literal": false,
+    "value": "name",
+    "kind": "Identifier",
+    "valueWithBar": "name"
+  },
+  "annotationAttachments": [],
+  "returnTypeNode": {
+    "position": {
+      "startColumn": 10,
+      "startLine": 3,
+      "endColumn": 16,
+      "endLine": 3
+    },
+    "id": "58f010ce-f978-4e85-ab5b-7228b9350937",
+    "symbolType": [
+      "null"
+    ],
+    "typeKind": "nil",
+    "grouped": false,
+    "nullable": false,
+    "kind": "ValueType"
+  },
+  "defaultableParameters": [],
+  "deprecatedAttachments": [],
+  "endpointNodes": [],
+  "body": {
+    "id": "1ae2339e-7148-4e46-bd18-e5a9788cc5f5",
+    "statements": [],
+    "kind": "Block"
+  },
+  "parameters": [],
+  "kind": "Function",
+  "public": false,
+  "private": false,
+  "remote": false,
+  "native": false,
+  "final": false,
+  "attached": false,
+  "lambda": false,
+  "worker": false,
+  "parallel": false,
+  "listener": false,
+  "deprecated": false,
+  "readonly": false,
+  "function_final": false,
+  "interface": false,
+  "required": false,
+  "record": false,
+  "compensate": false,
+  "abstract": false,
+  "optional": false,
+  "testable": false,
+  "client": false,
+  "resource": false,
+  "service": false,
+  "constant": false,
+  "allParams": []
+}
```

Comment:
Shall we keep a new line at the end of the file ?

Reply:
Added. These are generated jsons. for .ts files tslint should complain for missing new lines at eof.